pkgs/test/haskell/incremental/default.nix

```diff
@@ -0,0 +1,35 @@
+# Demonstration of incremental builds for Haskell. Useful for speeding up CI.
```

Comment:
Can we do btter than this? This test would succeed as well if the intermediates were just ignored.

Reply:
> The nice thing about incremental builds in general is that they degrade gracefully

Yeah, I was thinking about this as well, and this is the same conclusion I came to.  I couldn't think of a good way to test that the intermediates were actually being used, but this does test at least that the intermediate artifacts don't break the build.